edx_rest_framework_extensions/tests/test_permissions.py

```diff
@@ -155,3 +182,176 @@ def test_has_permission(self, enforce_scopes, authentication_class, jwt_filters,
+
+    def test_session_user_in_url_succeeds(self):
+        user = self._create_user()
+        request = self._create_request(username_in_url=user.username)
+        self._create_session(request, user)
+
+        response = self.SomeClassView().dispatch(request)
+        self.assertEqual(response.status_code, 200)
+
+    JwtTestCase = namedtuple('JwtTestCase', 'is_enforced, is_restricted, is_user_in_url, expected_response, expected_log')
+
+    @patch('edx_rest_framework_extensions.permissions.waffle.switch_is_active')
+    @patch('edx_rest_framework_extensions.permissions.log')
+    @ddt.data(
+        # **** Enforced ****
+        # unrestricted
+        JwtTestCase(is_enforced=True, is_restricted=False, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=True, is_restricted=False, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+
+        # restricted
+        JwtTestCase(is_enforced=True, is_restricted=True, is_user_in_url=True, expected_response=403, expected_log=JwtHasScopeLog),
+        JwtTestCase(is_enforced=True, is_restricted=True, is_user_in_url=False, expected_response=403, expected_log=JwtHasScopeLog),
+
+        # **** Unenforced ****
+        # unrestricted
+        JwtTestCase(is_enforced=False, is_restricted=False, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=False, is_restricted=False, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+
+        # restricted
+        JwtTestCase(is_enforced=False, is_restricted=True, is_user_in_url=True, expected_response=200, expected_log=JwtIsUnrestrictedLog),
+        JwtTestCase(is_enforced=False, is_restricted=True, is_user_in_url=False, expected_response=403, expected_log=IsUserInUrlLog),
+    )
+    @ddt.unpack
+    def test_jwt_cases(self, is_enforced, is_restricted, is_user_in_url, expected_response, expected_log, mock_log, waffle_mock):
+        waffle_mock.return_value = is_enforced
+        user = self._create_user()
+
+        auth_header = self._create_jwt_header(user, is_restricted=is_restricted)
+        request = self._create_request(
+            username_in_url=user.username if is_user_in_url else None,
+            auth_header=auth_header,
+        )
+
+        response = self.SomeClassView().dispatch(request)
+        self.assertEqual(response.status_code, expected_response)
+        self._assert_log(mock_log, expected_log)
+
+    JwtRestrictedTestCase = namedtuple('JwtRestrictedTestCase', 'scopes, filters, expected_response, expected_log')
+
+    @patch('edx_rest_framework_extensions.permissions.waffle.switch_is_active')
+    @patch('edx_rest_framework_extensions.permissions.log')
+    @ddt.data(
+        JwtRestrictedTestCase(scopes=['required_scope'], filters=['content_org:some_org'], expected_response=200, expected_log=JwtIsRestrictedLog),
```

Comment:
Maybe comments?
```
# Valid org filters
...

# Invalid org filters
...

# Invalid scopes
...
```

Reply:
no longer needed since we broke up into smaller test methods.